src/components/EmptyState/EmptyState.tsx

```diff
@@ -29,105 +30,113 @@ export interface EmptyStateProps {
   footerContent?: React.ReactNode;
 }
 
-export class EmptyState extends React.PureComponent<EmptyStateProps, never> {
-  render() {
-    const {
-      children,
-      heading,
-      image,
-      largeImage,
-      imageContained,
-      action,
-      secondaryAction,
-      footerContent,
-    } = this.props;
+export function EmptyState({
+  children,
+  heading,
+  image,
+  largeImage,
+  imageContained,
+  action,
+  secondaryAction,
+  footerContent,
+}: EmptyStateProps) {
+  return (
+    <WithinContentContext.Consumer>
+      {(withinContentContainer) => {
+        const className = classNames(
+          styles.EmptyState,
+          imageContained && styles.imageContained,
+          withinContentContainer
+            ? styles.withinContentContainer
+            : styles.withinPage,
+        );
 
-    const className = classNames(
-      styles.EmptyState,
-      imageContained && styles.imageContained,
-    );
+        const imageMarkup = largeImage ? (
+          <Image
+            alt=""
+            role="presentation"
+            className={styles.Image}
+            source={largeImage}
+            sourceSet={[
+              {source: image, descriptor: '568w'},
+              {source: largeImage, descriptor: '1136w'},
+            ]}
+            sizes="(max-width: 568px) 60vw"
+          />
+        ) : (
+          <Image
+            role="presentation"
+            alt=""
+            className={styles.Image}
+            source={image}
+          />
+        );
 
-    const imageMarkup = largeImage ? (
-      <Image
-        alt=""
-        role="presentation"
-        className={styles.Image}
-        source={largeImage}
-        sourceSet={[
-          {source: image, descriptor: '568w'},
-          {source: largeImage, descriptor: '1136w'},
-        ]}
-        sizes="(max-width: 568px) 60vw"
-      />
-    ) : (
-      <Image
-        role="presentation"
-        alt=""
-        className={styles.Image}
-        source={image}
-      />
-    );
+        const secondaryActionMarkup = secondaryAction
+          ? buttonFrom(secondaryAction, {plain: true})
+          : null;
 
-    const secondaryActionMarkup = secondaryAction
-      ? buttonFrom(secondaryAction, {plain: true})
-      : null;
+        const footerContentMarkup = footerContent ? (
+          <div className={styles.FooterContent}>
+            <TextContainer>{footerContent}</TextContainer>
+          </div>
+        ) : null;
 
-    const footerContentMarkup = footerContent ? (
-      <div className={styles.FooterContent}>
-        <TextContainer>{footerContent}</TextContainer>
-      </div>
-    ) : null;
+        const headingSize = withinContentContainer ? 'small' : 'medium';
```

Comment:
Maybe moving the DisplayText Size type to `src/types` to use here, there and in tests would be better.

Reply:
To clarify, you mean I should change `DisplayText.size` to use an enum instead of a union type so that I can use that there and here instead?

Reply:
(^ Just pushed up that change, but I think it might be a breaking change, at least for anyone consuming `@shopify/polaris` in a TypeScript project)

Reply:
I guess you are right, it's not worth the breaking change. Didn't realize it would be. We could just share the current type but not sure there's much value.